### PR TITLE
fix(learn): improve description of method in JavaScript

### LINKED ISSUE #55651: Pyramid generator - Step 26: references functions and objects before they are taught
### Describe the Issue

> A method in JavaScript is a function that's associated with certain values or objects.

The camper at this point has not been taught what a function, or an object, is.

Teaching methods before functions, and explaining methods by using a reference to functions and objects before they have been taught, seems backwards.

In order to understand what a method is, you must first understand what a function is, then you need to know what an object is, then you can learn that a method is a function on an object. You can not teach what a method is by referring to its constituent parts before they have been taught.

Step 47 teaches what a function is

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-introductory-javascript-by-building-a-pyramid-generator/step-47

I don't think objects ever show up in this challenge. It would be possible to convert the three top-level variables to properties on an object at some point between teaching functions and methods. Just as an option.

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-introductory-javascript-by-building-a-pyramid-generator/step-26

### Your code

```



```

### Expected behavior

Correct learning order.

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dejavuwl**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #55651 

Here is my first PR.

### CODE REVIEW

**Comment by ilenia-magoni** [comment]
This is not enough to solve the issue. It would need a big reordering of steps and topics

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/660f0a55847d6cc485f29ba5.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ dashedName: step-26
 
 # --description--
 
-A <dfn>method</dfn> in JavaScript is a function that's associated with certain values or objects. An example you've already encountered is the `.log()` method, which is part of the `console` object.
+An <dfn>object</dfn> is like a container that contains some data and operations that can be used to perform certain tasks. A <dfn>method</dfn> in JavaScript is a operation that's associated with certain values or objects. An example you've already encountered is the `.log()` method, which is part of the `console` object.
```
Comment (Sembauke) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/660f0a55847d6cc485f29ba5.md:10]:
```suggestion
A <dfn>function</dfn> is like a container that contains some data and operations that can be used to perform certain tasks. A <dfn>method</dfn> in JavaScript is a operation that's associated with certain values or objects. An example you've already encountered is the `.log()` method, which is part of the `console` object.
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

**Comment by dejavuwl** [comment]
I will try to make some reordering  to fix it later.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/660f0a55847d6cc485f29ba5.md (hunk 1 above)
Comment (ilenia-magoni) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-introductory-javascript-by-building-a-pyramid-generator/660f0a55847d6cc485f29ba5.md:10]: you never fixed this @dejavuwl

File: curriculum/i18n-curriculum
Comment (ilenia-magoni) [inline curriculum/i18n-curriculum:1]: this should not be in the PR

**Comment by ilenia-magoni** [comment]
closing as stale

**Pull request closed without merging**